smif/data_layer/__init__.py

```diff
@@ -1,3 +1,398 @@
+    def read_sos_model_runs(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_sos_model_run(self, sos_model_run):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_sos_models(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_sos_model(self, sos_model):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_sector_models(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_sector_model(self, sector_model_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_sector_model(self, sector_model):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_region_sets(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_region_set_data(self, region_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_interval_sets(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_interval_set_data(self, interval_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_units(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_region_set(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_interval_set(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_units(self, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_scenario_sets(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_scenarios(self, scenario_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_scenario_data(self, scenario_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_scenario_set(self, scenario_set):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_scenario(self, scenario):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_scenario_data(self, scenario_name, data):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_narrative_sets(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_narratives(self, narrative_set_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read_narrative_data(self, narrative_name):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_narrative_set(self, narrative_set):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_narrative(self, narrative):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def write_narrative_data(self, narrative_set_name, data):
+        raise NotImplementedError()
+
+
+class DatafileInterface(DataInterface):
+    """ Read and write interface to YAML / CSV configuration files
+    """
+    def __init__(self, config_path):
+        """Initialize file paths
```

Comment:
Don't use a docstring on the constructor - put this in the docstring for the class (including the argument) e.g.
```python
class DatafileInterface(DataInterface):
    """Read and write interface to YAML / CSV configuration files

    Arguments
    -----------
    config_path : str
        The path to the configuration folder
    """
```